integration/inspektor-gadget/integration_test.go

```diff
@@ -137,7 +134,8 @@ func testMain(m *testing.M) int {
 	initCommands := []*Command{}
 	cleanupCommands := []*Command{DeleteRemainingNamespacesCommand()}
 
-	if !*doNotDeployIG {
+	deployIG := !CheckNamespace("gadget")
+	if deployIG {
```

Comment:
I see a problem with this approach. If developers forget to remove IG before running the test by mistake, they can't easily realize that tests run with an "old" (and probably wrong) IG version. That's why I prefer the approach of failing the tests if IG is already installed unless developers manually specify `-no-deploy-ig`, which means that they are aware that they are running the tests with an "old" (and this time, probably correct) IG version.

Reply:
That is a fair point. How about printing a warning in case we use pre-installed components? If that isn't enough we can revert the `-no-deploy-ig/-no-deploy-spo` flags and only do this detection for `spo` since we don't really care of spo version? Should take care of this [issue](https://github.com/inspektor-gadget/inspektor-gadget/pull/1110#discussion_r1023830835) as well?

Reply:
oh, I use those flags a lot in my system to run integration tests locally. Can we add them back?

Reply:
Sorry about it. I am just testing my changes will open the PR soon.